packages/compiler/src/testing/types.ts

```diff
@@ -54,6 +54,8 @@ export interface TypeSpecTestLibrary {
 
 export interface TestHostConfig {
   libraries?: TypeSpecTestLibrary[];
+  /** the diagnostics will be ignored if this return false */
+  diagnosticFilter?: (diag: Diagnostic) => boolean;
```

Comment:
this is also a sign that the way hint was done is not working out for our tests, every single test host is adding it.